site/content/installation/installing-nic/deploy-with-nap-using-helm.md

```diff
@@ -0,0 +1,335 @@
+---
+title: Deploy NIC and WAF to Kubernetes using Helm
+toc: true
+weight: 500
+type: how-to
+product: NIC
+docs: DOCS-609
+---
+
+## How To: Build and Use a Local NGINX App Protect Docker Image to Compile WAF Policies and Integrate with NGINX Plus Ingress Controller
+
+This document outlines how to:
+
+- Set up license secrets to enable deployment into kubernetes
+- Use a WAF docker image to compile a policy JSON file into a compiled bundle
+- Set up PersistentVolumes that the deployed WAF instance can access to read the bundle from
+- Deploy NGINX Plus Ingress Controller with NAP in Kubernetes
+- Test example services to validate that the WAF policies work
+
+---
+
+## Step 1: Prepare Secrets and Credentials (From MyF5 Portal)
+
+Download the following files from your [MyF5 Portal](https://my.f5.com):
+
+- `nginx-repo.crt`
+- `nginx-repo.key`
+- `nginx-jwt.token`
+
+Store these files locally:
+
+```
+├── nginx-repo.crt
+├── nginx-repo.key
+└── nginx-repo.jwt
+```
+
+## Step 2: Pull the NGINX App Protect WAF Compiler image
+
+Log into the nginx private registry using your jwt file and the password `none` which you will have to type in when 
+asked:
+
+```bash
+$ docker login private-registry.nginx.com --username=$(cat nginx-repo.jwt)
+
+i Info → A Personal Access Token (PAT) can be used instead.
```

Comment:
The GitHub UI is eating it for the edit suggestion I'm trying to make:

We use `shell` over `bash`: both work, but the former is more generic.

This is more about consistency than anything else: I use zsh, but it's POSIX compliant and we don't anticipate a user would be mislead into a particular syntax consuming the Markdown view of documentation than the actual website.

Reply:
I've changed it to `shell` in this document, however there are a bunch more in this repository still using `bash` as a code type

Reply:
Yeah, I know.  A find and replace pass would fix all of it: I generally fix small stuff like this as a go.

I might do it as its own PR in the future.

Reply:
All right, let's fix up the rest in a separate PR. Marking this as resolved.